### PR TITLE
Implements a new view for an expanded finding aid collection

### PR DESCRIPTION
**Pull request by jrgriffiniii**
Advances #89, but I would like to first please request some feedback before finalizing this.

### CODE REVIEW

**Comment by jrgriffiniii** [comment]
There was a bug with this originally, and only a limited number of child documents were rendered. Also, the Table of Contents widget was not expanded:

image

**Comment by jrgriffiniii** [comment]
This needs to be rebased.

**Review by eliotjordan — COMMENTED** [review]
@jrgriffiniii A few thoughts / questions:

1. We will need an expanded view button on the TOC header bar to be compliant with the wireframes.
1. The component titles should be links that take the user to the component. Right?
1. I wonder about expanding the TOC. The wireframe has it closed and it just seems like unnecessary and duplicative information to me.
1. From the wireframes, it looks like we need more metadata for components at a certain level. Looks like "Scope note" in the drawings. I'm not sure if that's correct or just filler.

**Comment by jrgriffiniii** [comment]
https://github.com/pulibrary/pulfalight/pull/435/commits/59ff113557183ebde1fd87dda4717177b41a78d7 introduces the metadata for components with `scope note` and `arrangement`, as well as an `expand` and `collapse` button:
image

image

**Comment by jrgriffiniii** [comment]
https://github.com/pulibrary/pulfalight/pull/435/commits/9f766d2b47b2841b75bd4157b001ade966ca4bda ensures that component titles are links and improves the styling for the extent badges:

image

**Comment by jrgriffiniii** [comment]
https://github.com/pulibrary/pulfalight/pull/435/commits/cbe210e9f31b57a09ca56e18c883eeb40534727f ensures that the Table of Contents component is collapsed for the expanded Collection View.

**Comment by jrgriffiniii** [comment]
Rebasing this introduced a number of test failures which I'm now having difficulty reproducing locally. I suspect that this might be due to some updates applied to the EAD XML fixtures used during some the tests.

**Review by eliotjordan — CHANGES_REQUESTED** [review]
@jrgriffiniii  I have one code comment so far and a serious concern about performance. After experimenting a little, it seems that the SolrDocumentTree is making a single solr query for each child document. This works OK in dev with fixtures, but caused a serious problem on staging when trying to build the tree for the full Toni Morrison collection. The way I solved this for the TOC was to make one Solr request for the entire collection document with all of it's nested children and then parse that in the Ruby.

File: spec/models/solr_document_tree_spec.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  let(:values) do
+    indexer.map_record(record)
+  end
+  let(:solr_document) do
+    SolrDocument.new(values)
+  end
+
+  context "when the Solr Document contains one or more member components" do
+    let(:fixture_path) do
+      Rails.root.join("spec", "fixtures", "aspace", "generated", "publicpolicy", "MC152.EAD.xml")
```
Comment (eliotjordan) [inline spec/models/solr_document_tree_spec.rb:39]: No need to index this record again. It's already a fixture. On line 34, you could just use `SolrDocument.find("MC152")`

**Comment by jrgriffiniii** [comment]
https://pulfalight-staging.princeton.edu/catalog/MC120?expanded=true is still quite slow even with https://github.com/pulibrary/pulfalight/pull/435/commits/0b2440f1772e7d1c14178d5c747e17ceb409fe9d, but there were definitely improvements. My suspicion is that this might also just be traversing of the nested Solr Documents.

**Review by eliotjordan — APPROVED** [review]

**Pull request merged**